File: java/TwoSum.java
import java.util.Arrays;
class TwoSum {
  public static int[] twoNumberSum(int[] array, int targetSum) {
    // Write your code here.
		Arrays.sort(array);
		int start = 0;
		int end = array.length - 1;
		while (start < end){
			int sum = array[start] + array[end];
			if (sum == targetSum){
      	return new int[] {array[start], array[end]};
			} else if (sum > targetSum){
      	end--;
    	} else if (sum < targetSum){
      	start++;
    	}
		}
  	return new int[0];
  }
}

File: javascript/twoSum.js
function twoSum(array, targetSum) {
  // Write your code here.
  let result = [];
  array.sort((a, b) => a - b);
  // console.log(array)
  let start = 0;
  let end = array.length - 1;
  while (start < end) {
    if (array[start] + array[end] === targetSum) {
      result.push(array[start]);
      result.push(array[end]);
    }
    if (array[start] + array[end] >= targetSum) {
      end--;
    } else if (array[start] + array[end] <= targetSum) {
      start++;
    }
  }
  return result;
}

console.log(twoSum([2, 5, 5, 11], 10));
